### PR TITLE
Split tests in nft_test.ts to 2 modules: nft_test.rs and meta_loader_test.rs.

### PR DESCRIPTION
**Pull request by Selarion**

### CODE REVIEW

File: client/tests/nft_test.rs [hunk 1]
```diff
@@ -18,7 +24,7 @@ async fn test_get_nft_data() {
 }
 
 #[tokio::test]
-async fn test_get_nft_collection_data() -> anyhow::Result<()> {
+async fn test_get_collection_data() -> anyhow::Result<()> {
```
Comment (dbaranovstonfi) [inline client/tests/nft_test.rs:27]: please revert to `test_get_nft_collection_data`

File: client/tests/nft_test.rs [hunk 2]
```diff
@@ -40,78 +46,65 @@ async fn test_get_nft_address_by_index() -> anyhow::Result<()> {
...
-}
 
-#[tokio::test]
-async fn test_get_nft_content_arkenston() -> anyhow::Result<()> {
-    common::init_logging();
-    let client = common::new_mainnet_client().await;
-    let factory = assert_ok!(TonContractFactory::builder(&client).build().await);
-    let contract = factory.get_contract(&assert_ok!(
-        "EQDhR36C8pSVtyhOFtE9nh2DFq4WYUbTZFmvjfnShlrXq2cz".parse()
+    let expected_collection_address = assert_ok!(TonAddress::from_base64_url(
```
Comment (dbaranovstonfi) [inline client/tests/nft_test.rs:60]: please check this construction with clippy. no need to convert `&'static str` to String and then back to `&str`. Also method TonAddress::parse()  might be handy

File: client/tests/nft_test.rs [hunk 3]
```diff
@@ -40,78 +46,65 @@ async fn test_get_nft_address_by_index() -> anyhow::Result<()> {
...
-    let res = assert_ok!(contract.get_nft_data().await);
-    let meta_loader = assert_ok!(NftItemMetaLoader::default());
-    let content_res = assert_ok!(meta_loader.load(&res.individual_content).await);
-    assert_eq!(
-        content_res.image.unwrap(),
-        "https://static.ston.fi/stake-nft/i4.jpg"
-    );
-    assert_eq!(content_res.name.unwrap(), "ARKENSTON NFT");
+    let expected_owner_address = assert_ok!(TonAddress::from_base64_url(
+        &"EQCgjHh831e9_qlCWLgaAwEIQ8qOolUT831vJF0bau6LMV5G".to_string()
```
Comment (dbaranovstonfi) [inline client/tests/nft_test.rs:64]: see comment above

File: client/tests/nft_test.rs [hunk 4]
```diff
@@ -120,83 +113,44 @@ async fn test_get_nft_collection_content_uri() -> anyhow::Result<()> {
...
 #[tokio::test]
-async fn test_get_nft_collection_content_some() -> anyhow::Result<()> {
+async fn test_get_nft_address_by_index_is_valid() -> anyhow::Result<()> {
     common::init_logging();
     let client = common::new_mainnet_client().await;
-    let factory = assert_ok!(TonContractFactory::builder(&client).build().await);
+    let factory = TonContractFactory::builder(&client).build().await?;
     let contract = factory.get_contract(&assert_ok!(
-        "EQCbOjwru5tBb2aaXZEHbiTCVIYQ6yDNAe8SSZkP4CozibHM".parse()
+        "EQB2iHQ9lmJ9zvYPauxN9hVOfHL3c_fuN5AyRq5Pm84UH6jC".parse()
```
Comment (dbaranovstonfi) [inline client/tests/nft_test.rs:136]: I wonder why the address of contract has been changed. 
 Looks like the order of tests also has been changed. We should avoid mixing up test order is possible.

File: client/src/contract/nft/item_contract.rs [hunk 5]
```diff
@@ -81,6 +81,7 @@ pub trait NftItemContract: TonContractInterface {
     /// the individual content of this NFT item and
     /// returns the full content of the NFT item in format
     /// that complies with standard TEP-64.
+    // TODO Add tests to this method
```
Comment (dbaranovstonfi) [inline client/src/contract/nft/item_contract.rs:84]: Lets do it in the same scope

**Review by dbaranovstonfi — CHANGES_REQUESTED** [review]

File: client/tests/nft_test.rs (hunk 1 above)
Comment (Selarion) [reply client/tests/nft_test.rs:27]: Name of tested method: "get_collection_data()". So i suggest we keep this name or make ".test_nft_get_collection_data()"

File: client/tests/nft_test.rs (hunk 4 above)
Comment (Selarion) [reply client/tests/nft_test.rs:124]: It looks like a random git diff. I checked, and this address is correct and the same as the address before the refactor. You can check this in test 'test_get_nft_address_by_index' (line: 43 of this file).

File: client/tests/nft_test.rs (hunk 2 above)
Comment (Selarion) [reply client/tests/nft_test.rs:60]: Updated. I forgot to run cargo clippy with the --test flag.

File: client/tests/meta_loader_test.rs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        attributes: None,
+    };
+
+    let res = assert_ok!(meta_loader.load(&content).await);
+    assert_eq!(expected_res, res);
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_load_item_metadata_arkenstone() -> anyhow::Result<()> {
```
Comment (dbaranovstonfi) [inline client/tests/meta_loader_test.rs:37]: Please update the name to test_load_item_metadata_arkenston

**Review by dbaranovstonfi — COMMENTED** [review]
Just one last comment

**Review by dbaranovstonfi — APPROVED** [review]

**Pull request merged**